scio-elasticsearch/es7/src/main/scala/com/spotify/scio/elasticsearch/IndexAdmin.scala

```diff
@@ -78,4 +89,134 @@ object IndexAdmin {
+   */
+  private def removeIndex(
+    esOptions: ElasticsearchOptions,
+    index: String,
+    client: IndicesClient,
+    timeout: TimeValue
+  ): AcknowledgedResponse = {
+    val request = new DeleteIndexRequest(index)
+      .timeout(timeout)
+
+    client.delete(request, RequestOptions.DEFAULT)
+  }
+
+  /**
+   * Add index alias and remove the alias from all other indexes if it is already pointed to any.
+   * If index already exists or some other error occurs this results in a [[scala.util.Failure]].
+   *
+   * @param alias        to be re-assigned
+   * @param indexName index to point the alias to
+   * @param timeout defaults to 1 minute
+   */
+  def createOrUpdateAlias(
+    nodes: Iterable[HttpHost],
+    alias: String,
+    indexName: String,
+    removePrevious: Boolean,
+    timeout: TimeValue = TimeValue.timeValueMinutes(1)
+  ): Try[AcknowledgedResponse] = {
+    val esOptions = ElasticsearchOptions(nodes.toSeq)
+    createOrUpdateAlias(esOptions, alias, indexName, removePrevious, timeout)
+  }
+
+  /**
+   * Add index alias with an option to remove the alias from all other indexes if it is already
+   * pointed to any.
+   * If index already exists or some other error occurs this results in a [[scala.util.Failure]].
+   *
+   * @param alias            to be re-assigned
+   * @param indexName        index to point the alias to
+   * @param removePrevious   When set to true, the indexAlias would be removed from all indices it
+   *                         was assigned to before adding new index alias assignment.
+   */
+  def createOrUpdateAlias(
+    esOptions: ElasticsearchOptions,
+    alias: String,
+    indexName: String,
+    removePrevious: Boolean,
+    timeout: TimeValue
+  ): Try[AcknowledgedResponse] =
+    indicesClient(esOptions)(client =>
+      createOrUpdateAlias(esOptions, alias, indexName, removePrevious, client, timeout)
+    )
+
+  /**
+   * Add index alias and remove the alias from all other indexes if it is already pointed to any.
+   * If index already exists or some other error occurs this results in a [[scala.util.Failure]].
+   *
+   * @param alias            to be re-assigned
+   * @param indexName        index to point the alias to
+   * @param removePrevious   When set to true, the indexAlias would be removed from all indices it
+   *                         was assigned to before adding new index alias assignment
+   *
+   */
+  private def createOrUpdateAlias(
+    esOptions: ElasticsearchOptions,
+    alias: String,
+    indexName: String,
+    removePrevious: Boolean,
+    client: IndicesClient,
+    timeout: TimeValue
+  ): AcknowledgedResponse = {
+
+    val getAliasesResponse =
+      client.getAlias(new GetAliasesRequest(alias), RequestOptions.DEFAULT)
+
+    val request = new IndicesAliasesRequest()
+      .addAliasAction(
+        new AliasActions(AliasActions.Type.ADD)
+          .index(indexName)
+          .alias(alias)
+      )
```

Comment:
What I meant with `isWriteIndex: Boolean` was smth along the lines of:

```suggestion
    // indices: Iterable[(String, Boolean)] (index name, is write index)
    require(
      indices.find(_._2).size == 1,
      "Only one index per alias can be assigned to be the write index at a time"
    
    val request = indices.foldLeft(new IndicesAliasesRequest()) {
      case (request, (idx, isWriteIndex)) =>
        request.addAliasAction(
          new AliasActions(AliasActions.Type.ADD)
            .index(idx)
            .writeIndex(isWriteIndex)
            .alias(alias)
        )
    }
```